### DISCUSSION TITLE
What would it take to compile this for WebAssembly? Remote Browser Access.

### DISCUSSION THREAD

**Discussion by ghost** [neovide/neovide#2728] [Ideas]
I have a bit of a rare use case for Neovim. 
I want to be able to access my neovim running on my home computer from work, but I can't install anything on my work system and I don't have a laptop.
Is it possible to 
1. Run neovim in the browser.
2. Connect to a headlesss neovim session running on a entirely different system through ssh perhaps?

Just as a note I am not asking for this to be made I am interested in what would be required to make it as I might invest in making it myself.

**Comment by fredizzimo** [comment]
It's not totally out of questions. [Winit](https://github.com/rust-windowing/winit/issues), the main platform library we use supports the web. 

And we are also in the progress of moving our renderer to wgpu through [Vide](https://github.com/neovide/vide). So, after that I think it could be done with relatively low effort.

The major obstacle left after that is that currently Neovide does not work well on medium to high latency links. Basically, if the ping is higher than the keyboard repeat rate usually around 30 ms, it becomes totally unusable. The reason for that is that the communication library we use [nvim-rs](https://github.com/killthemule/nvim-rs) does not really support RPC notifications, which means that all the inputs have to travel a full roundtrip to Neovim and back, and in order. This is something that can and should be fixed in any case.

BTW, this is probably not what you are looking for, but Neovim can run in the browser in edit boxes with [firenvim](https://github.com/glacambre/firenvim)

**Reply by ghost** [reply]
I see, thank you very much for the detailed response!
Network definitely seems like the limiting factor here and I know for myself my home network most likely would not support that kind of low latency connection. 
This is a crazy over engineered personal idea but I could host my Neovim headless on a Edge server to get the lowest possible latency.
Definitely would be amazing to see Neovide for the web though!

**Comment by unphased** [comment]
I came here to look for SSH support for neovide. I just tried out neovide and it's blowing my mind completely. My very seriously customized neovim setup runs really well in neovide. The animations are really chef's kiss here to the point where i may have to put all of my terminal emulator, tmux etc customization efforts on hold, as i'm not sure I could stomach going back... 

I'm curious to know a little bit more technically about the nvim-rs limitation. Gotta pore over the long list of features that need to be built up or worked around. 

In broad strokes: 
1. remote (aka neovide <--> ssh <--> nvim)
2. image rendering (a topic that got me down this path i'm on today of evaluating terminal/neovim related dev workflow technologies in 2024)
3. terminal (been driving alacritty for 10 years or so, quite open to moving away)
4. multiplexing (been driving tmux for 15 years or so, again quite open to moving away)

Looks like (1) has plans (maybe) as you touched upon earlier, (2) may be easy to get working, not sure (3) and (4) have much hope realistically in the short term. But what I can say is neovide shows what is actually possible, and a good experience in terminals in general and for terminal multiplexing definitely look to be possible. All it really needs to support is animating stuff scrolling up (and back down when browsing scrollback). 

anyway i guess my dreams of having a single workflow i can build to use from inside a terminal emulator is getting torpedoed. maybe i was subconsciously procrastinating trying out neovide because of this. either way... no regrets...
